packages/blockchain-wallet-v4-frontend/src/data/components/recoveryPhrase/reducers.ts

```diff
@@ -0,0 +1,33 @@
+import * as AT from './actionTypes'
+import { RecoveryPhraseActionTypes, RecoveryPhraseState } from './types'
+
+const INITIAL_STATE: RecoveryPhraseState = {
+  step: 'RECOVERY_PHRASE_INTRO'
+}
+
+export function recoveryPhraseReducer (
+  state = INITIAL_STATE,
+  action: RecoveryPhraseActionTypes
+): RecoveryPhraseState {
+  if (action.type === AT.SET_STEP) {
+    switch (action.payload.step) {
+      case 'RECOVERY_PHRASE_INTRO':
+      case 'FIRST_SET_WORDS':
+      case 'SECOND_SET_WORDS':
+      case 'CONFIRM_WORDS':
+      case 'CONFIRM_WORDS_SUCCESS':
+        return {
+          ...state,
+          step: action.payload.step
+        }
+      default: {
+        return {
+          ...state,
+          step: action.payload.step
+        }
+      }
+    }
```

Comment:
For borrow and sb there were other state props we were changing depending on the step. I don't think you need lines 14-23, you can just change the step.